### ISSUE TITLE
TypeError: iV is not a constructor

### ISSUE THREAD

**Issue by ChrisB1123** [nextauthjs/next-auth#10092]
### Environment

  System:
    OS: macOS 11.7.3
    CPU: (8) x64 Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz
    Memory: 172.39 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 21.3.0 - ~/.nvm/versions/node/v21.3.0/bin/node
    npm: 10.4.0 - ~/.nvm/versions/node/v21.3.0/bin/npm
  Browsers:
    Brave Browser: 113.1.51.118
    Chrome: 121.0.6167.184
    Safari: 16.3
  npmPackages:
    @auth/prisma-adapter: ^1.0.9 => 1.4.0 
    next: 14.1 => 14.1.0 
    next-auth: 5.0.0-beta.13 => 5.0.0-beta.13 
    react: ^18 => 18.2.0 

### Reproduction URL

https://github.com/ChrisB1123/auth-bug

### Describe the issue

This occurs only in production.

If I have my AUTH_URL environment variable set to {host}/api/auth, I receive the below error when navigating to /dashboard in the example code. /dashboard is protected via middleware, so the error occurs when being redirected to /signin

```
2024-02-21T08:35:32.430677+00:00 app[web.1]:  ⨯ TypeError: iV is not a constructor
2024-02-21T08:35:32.430747+00:00 app[web.1]:     at iX (/app/.next/server/app/api/auth/[...nextauth]/route.js:357:50464)
2024-02-21T08:35:32.430748+00:00 app[web.1]:     at t (/app/.next/server/app/api/auth/[...nextauth]/route.js:357:56404)
2024-02-21T08:35:32.430748+00:00 app[web.1]:     at /app/node_modules/next/dist/compiled/next-server/app-route.runtime.prod.js:6:42490
2024-02-21T08:35:32.430749+00:00 app[web.1]:     at /app/node_modules/next/dist/server/lib/trace/tracer.js:133:36
2024-02-21T08:35:32.430749+00:00 app[web.1]:     at NoopContextManager.with (/app/node_modules/next/dist/compiled/@opentelemetry/api/index.js:1:7062)
2024-02-21T08:35:32.430750+00:00 app[web.1]:     at ContextAPI.with (/app/node_modules/next/dist/compiled/@opentelemetry/api/index.js:1:518)
2024-02-21T08:35:32.430750+00:00 app[web.1]:     at NoopTracer.startActiveSpan (/app/node_modules/next/dist/compiled/@opentelemetry/api/index.js:1:18093)
2024-02-21T08:35:32.430750+00:00 app[web.1]:     at ProxyTracer.startActiveSpan (/app/node_modules/next/dist/compiled/@opentelemetry/api/index.js:1:18854)
2024-02-21T08:35:32.430751+00:00 app[web.1]:     at /app/node_modules/next/dist/server/lib/trace/tracer.js:122:103
2024-02-21T08:35:32.430751+00:00 app[web.1]:     at NoopContextManager.with (/app/node_modules/next/dist/compiled/@opentelemetry/api/index.js:1:7062)
```

If I remove the AUTH_URL environment variable entirely, I'm redirected to /signin successfully, but the callbackUrl option is localhost (and I'm not sure where the port comes from)
`?callbackUrl=https%3A%2F%2Flocalhost%3A35891%2Fdashboard`

### How to reproduce

1. Clone auth-bug repo
2. Set database url & auth secret environment variables
3. Set up Heroku instance with database attached
4. Run npx prisma generate & npx prisma db push
5. Push code to Heroku
6. Navigate to {heroku_url}/dashboard
7. Check logs

### Expected behavior

Either no error should be received when my AUTH_URL is set to {host}/api/auth, or when I don't set AUTH_URL, then it should correctly pick up the callback url.

**Comment by ChrisB1123** [comment]
Resolved by reverting to next-auth v5.0.0-beta.4

Which also required pegging to @auth/core v0.18.4 and @auth/prisma-adapter v1.0.9

I've left the public repo so the versions in there should still cause the error

**Comment by prim1013** [comment]
Seems related to [this PR](https://github.com/nextauthjs/next-auth/pull/9795)

**Comment by balazsorban44** [comment]
Duplicate of #9922. This was an upstream Next.js issue, which was already fixed in a `canary` release. Please upgrade to `next@canary`.

https://github.com/nextauthjs/next-auth/issues/9922#issuecomment-1930121175

**Comment by jrhizor** [comment]
In case anyone runs into this in the future, the fix is included in `next@14.1.1` and after.
